File: java/gigasecond/src/main/java/Gigasecond.java
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;

class Gigasecond {

    private static final long GIGA = 1_000_000_000;
    private final LocalDateTime dateTime;    
        
    Gigasecond(LocalDate date) {
        this(date.atTime(0, 0));
    }

    Gigasecond(LocalDateTime dateTime) {
        this.dateTime = dateTime.plusSeconds(GIGA);
    }

    LocalDateTime getDateTime() {
        return dateTime;
    }
}

File: typescript/gigasecond/gigasecond.ts
const GIGA: number = 1e9;
const SECOND: number = 1e3;
const GIGASECOND: number = GIGA * SECOND;

export default class Gigasecond {
    constructor(private readonly start: Date) {

    }
    date(): Date {
        return new Date(this.start.getTime() + GIGASECOND);
    }
}
